provisioning/WFLY-21122-WildFly-Feature_pack-Family.adoc

```diff
@@ -0,0 +1,238 @@
+Feature pack authors who want to allow selection of a dependency FP via a criteria must discuss with 
+the maintainers of the feature packs that advertise that criteria whether their feature packs are suitable for use as a dependency.
+
+=== Implementation notes
+
+The changes in the WildFly project are:
+
+* Bump to Galleon 7.x and WildFly Galleon Plugins 8.1.x
+* Update the three `wildfly-feature-pack-build.xml` files to define their family and criteria.
+* Update the WildFly full `wildfly-feature-pack-build.xml` file to inherit criteria from its WildFly EE dependency.
+
+`wildfly-feature-pack-build.xml` extract:
+
+```
+<build xmlns="urn:wildfly:feature-pack-build:3.5" producer="wildfly@maven(org.jboss.universe:community-universe):current">
+    <family name="wildfly">
+      <criteria name="microprofile-7.1+"/>
+      <criteria name="microprofile-7.1"/>
+    </family>
+    <dependencies>
+        <dependency group-id="org.wildfly" artifact-id="wildfly-ee-galleon-pack" allowedFamily="wildfly:jakarta-ee-deployment,jakarta-ee-10+">
+    ...
+</build>
+```
+
+The `allowedFamily` attribute defines the criteria that are inherited from `wildfly-ee-galleon-pack`. We can see that `jakarta-ee-11` is not inherited.
+This is a choice to not bound WildFly feature-pack to a specific version of Jakarta EE. This choice could be revisited.
+
+An example of a feature-pack that depends on WildFly EE 10:
+
+```
+<build xmlns="urn:wildfly:feature-pack-build:3.5" producer="org.example:myfeaturepack">
+    <dependencies>
+        <dependency group-id="org.wildfly" artifact-id="wildfly-ee-galleon-pack" allowedFamily="wildfly:jakarta-ee-10">
+    <dependencies>
+    ...
+</build>
+```
+
+The `allowedFamily` attribute contains the `<family>:[comma separated list of criterion]`. It is expected that the dependency be part of the 
+mentioned family and implements the listed criteria. Any error must be reported at feature-pack build time.
+
+
+==== Potential impact for WildFly Glow
+
+Today WildFly Glow CLI and its integration in the WildFly Maven Plugin offer the `wildfly-preview` option to contextualize the feature-packs discovery to use WildFly Preview feature-packs. This option should be deprecated
+and a new option `wildfly-variant=<variant name>` should be introduced. An immediate variant is `preview`, but new ones could be introduced in the future.
+
+=== Non-Requirements
+
+* It is not a requirement that the same set of exposed criteria is present for each new WildFly Major releases.
+* It is not a requirement that WildFly not introduce incompatible changes in its contracts in a new Major release. 
+This feature does not introduce new compatibility guarantees across releases beyond what WildFly already provides.
+
+=== Future Work
+
+* Introduce an 'or' operator for criteria that you are expecting from a dependency. Something like: `wildfly:jakarta-ee10|jakarta-ee11,...`
```

Comment:
```suggestion
* Introduce an 'or' operator for criteria that you are expecting from a dependency. Something like: `wildfly:jakarta-ee10|jakarta-ee11,...`. Feature packs will rely on dependents using that instead of providing criteria like `jakarta-ee10+`
```

Reply:
Ok, added.